ledger-core/virtual/handlers/vstatereport.go

```diff
@@ -60,14 +60,86 @@ func (s *SMVStateReport) Init(ctx smachine.InitializationContext) smachine.State
 	}
 
 	ctx.SetDefaultMigration(s.migrationDefault)
-	return ctx.Jump(s.stepProcess)
+	return ctx.Jump(s.stepValidate)
 }
 
 func (s *SMVStateReport) migrationDefault(ctx smachine.MigrationContext) smachine.StateUpdate {
 	ctx.Log().Trace("stop processing VStateReport since pulse was changed")
 	return ctx.Stop()
 }
 
+func (s *SMVStateReport) stepValidate(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	switch report := s.Payload; {
+	case report.GetObject().IsEmpty():
+		return ctx.Error(throw.New("Object should not be empty"))
+	case report.GetAsOf().IsUnknown():
+		return ctx.Error(throw.New("AsOf can not be Unknown"))
+	case !report.GetDelegationSpec().IsZero():
+		return ctx.Error(throw.New("DelegationSpec should be empty"))
+	default:
+		switch report.GetStatus() {
+		case payload.Unknown:
+			return ctx.Error(throw.New("Unexepected Unknown status received"))
+		case payload.Missing:
+			if err := validateZeroPending(report); err != nil {
+				return ctx.Error(err)
+			}
+
+			if err := validateEmptyState(report); err != nil {
+				return ctx.Error(err)
+			}
+		case payload.Empty:
+			if report.GetOrderedPendingCount() != 1 {
+				return ctx.Error(throw.New("Should be one ordered pending"))
+			}
+
+			objectPulse := report.GetAsOf()
+			currPulse := s.pulseSlot.PulseNumber()
+			orderedPendingPulse := report.GetOrderedPendingEarliestPulse()
+
+			if orderedPendingPulse < objectPulse || currPulse < orderedPendingPulse {
```

Comment:
`orderedPendingPulse < objectPulse` looks incorrect, I expect `<=` here

Reply:
I don't understand the second condition either, why `current pulse` is less than `ordered pending pulse`?

Reply:
object<=pending<cur this is our condition that we expected, my condition is reverse